### ISSUE TITLE
Deployed contract via Ubuntu but transactions not updating

### ISSUE THREAD

**Issue by abu9725** [FuelLabs/fuels-wallet#993]
### What version of Wallet are you using?

0.13.10

### What is happening?

Deployed contracts and interactions on the network are not updating in wallet. No interactions from wallet go through to the network. see below smart contract deployed via ubuntu:
Network: https://beta-4.fuel.network
Contract ID: 0xe563fe142844a0518a4dbf90b1a30313ca070980d9cbd3d42f85dddabc3cca0c
Deployed in block 0x115416bec5b99797a7f0db2a17400cb57830be07f8c3dbe97b64931f14787958
Deployer address: fuel1srgy7l3gl8jjrst3tz0lgtsnut0vn6gdqjq3rgngmnuxemq5kawqzzgxgy 
<img width="1121" alt="Screenshot 2023-12-05 at 20 59 57" src="https://github.com/FuelLabs/fuels-wallet/assets/115375966/38e90d55-caf0-48d4-9a4d-66708f5f3d4d">
<img width="346" alt="Screenshot 2023-12-05 at 20 59 32" src="https://github.com/FuelLabs/fuels-wallet/assets/115375966/18622ea6-bd3e-4f8d-8342-6eb7fecbc986">

### steps to reproduce

Use same wallet version and try to deploy a smart contract or send tokens from wallet- got a fail

### How it should behave?

interactions should go through to the network
use ubuntu to deploy and it works

### What os are you seeing the problem on?

Mac/Apple Desktop

### What browser?

Chrome

### What is your browser version?

113.0.5672.63 (Official Build) (arm64)

**Comment by luizstacio** [comment]
Hi @abu9725, are this still happening? Can you check if your Wallet has the correct network selected?

**Comment by abu9725** [comment]
Hi Luiz

Just tested again with a new contract:
 Contract counter-contract Deployed!

Network: https://beta-4.fuel.network
Contract ID: 0x4291ee5fdee915a7889346d546472c2be0da4e3db98bd49bf9d19ec35d6f8eb5
Deployed in block 0x27908e6c910ff49f80b4838268623128a7ee0dbc49a2afe45c4fab4e9cb49edf

attached result from transaction history viewer on fuel wallet
<img width="351" alt="Screenshot 2024-01-10 at 21 48 25" src="https://github.com/FuelLabs/fuels-wallet/assets/115375966/cbf33435-a949-42ea-835c-b62ee34ec151">
 , doesn't show up!

but it is showing as transaction on fuelstat.xyz
<img width="1143" alt="Screenshot 2024-01-10 at 21 52 50" src="https://github.com/FuelLabs/fuels-wallet/assets/115375966/0df699bc-4fa0-4c7e-850b-4cd1203e25a9">


so in conclusion yes the wallet is beta testnet4 and the contract also deployed to same network. looks like faulty interface on wallet extension probably you are already aware but just thought to flag this up.

thanks and regards
Abu

**Comment by luizstacio** [comment]
@abu9725 I was able to reproduce the issue and create a patch, for it. We are currently working on a update for next versions of the fuel-core and this fix should be included on it.

The issue happens because your contract creation Transactions do not include any  Input of type Coin, only Message. And the SDK was throwing a error as it was trying to access data on a input coins that do not exist on your tx inputs.

As soon this is release, I'm going to update here.

**Comment by luizstacio** [comment]
@abu9725 this issue has been fixed by https://github.com/FuelLabs/fuels-wallet/pull/1041 and will be release on version v0.14.2 of the Fuel Wallet and Fuel Wallet Development. You can also test installing from source code. [here](https://next-wallet.fuel.network/docs/install/#install-from-source-code).

**Comment by abu9725** [comment]
Great, glad this could be sorted - hope it was useful for the team!
